packages/application-shell/src/components/handle-apollo-errors/handle-apollo-errors.js

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { wrapDisplayName } from 'recompose';
+import { ReactReduxContext } from 'react-redux';
+import { handleActionError } from '@commercetools-frontend/actions-global';
+
+function handleApolloErrors(queryResultNames) {
```

Comment:
Maybe `withApollorErrorHandling`?

Reply:
I am not sure Apollo calls these `queryResultNames` maybe just `queryNames`?

Reply:
`withApollorErrorHandling` usually implies that the HOC injects props into the wrapped component, but in this case we only have a side effect.

Reply:
i also like `withApolloErrorHandling` or even `injectApolloErrorHandling`

Reply:
> I am not sure Apollo calls these `queryResultNames` maybe just `queryNames`?

They call it `QueryResult`, see https://www.apollographql.com/docs/react/api/react-apollo.html#query-render-prop

Reply:
I agree that `with` and `inject` should be reserved for component which add an additional prop (inject) into a component. Recompose also has a bunch of components `getContext` or `defaultProps` etc which are named without. I think `handleApolloErrors` is a good name after all. Sorry for the confusion.